### ISSUE TITLE
KnpMenuBundle menu renderer throws an exception if it is used together with MenuBundle

### ISSUE THREAD

**Issue by nettiopsu** [symfony-cmf/MenuBundle#242]
If some is using MenuBundle and KnpMenuBundle at the same time and will try to use the following string:
`{{ knp_menu_render('MyOwnMainBundle:Builder:mainMenu'}) }}`

Symfony CMF MenuBundle will throw then an error "Unknown namespace prefix(es)" in a Twig template for the string "MyOwnMainBundle:Builder:mainMenu". Though expected behavior is to skip this name as non existent and pass the work to KnpMenuBundle, which knows how to handle it.

My guess is that PhpcrUtils that are used in MenuBundle got a new throwable exception, which is not handled yet by this it. The exception is thrown by `has` function (that should not throw any exception, but return false in this case), which is using `find` function that has try-catch blocks for RepositoryException and PathNotFoundException, but not for NamespaceException:
```php
public function has($name, array $options = array())
{
 return $this->find($name, $options, false) instanceof NodeInterface;
}
protected function find($name, array $options, $throw)
{
...
```
https://github.com/symfony-cmf/MenuBundle/blob/master/Provider/PhpcrMenuProvider.php

And here is where this exception comes from (line 68):
```php
throw new NamespaceException(sprintf('Unknown namespace prefix(es) (%s) in path %s', implode(' and ', $unknown), $path));
```
https://github.com/phpcr/phpcr-utils/blob/master/src/PHPCR/Util/PathHelper.php

**Comment by dbu** [comment]
indeed that looks like an unintended behaviour. on which line of the
PhpcrMenuProvider is the exception triggered? i would have expected this
to happen in
https://github.com/symfony-cmf/MenuBundle/blob/master/Provider/PhpcrMenuProvider.php#L223-L231
and NamespaceException should be instanceof RepositoryException. does
the exception only happen later? then it feels like a bug in
PathHelper::assertValidAbsolutePath as the path surely is not valid.

could you do a pull request with a test that tries to load this menu?
the test would fail and show what exactly happens.

**Comment by nettiopsu** [comment]
Ok, sorry, it seems that you really have fixed this in master branch, but I am using the last stable version 1.2.0
The problem is that the last stable version of phpcr-utils (1.2.8, which allowed to be used by doctrine/phpcr-bundle, doctrine/phpcr-odm and jackalope/jackalope) contains already this new NamespaceException, which is unhandled by MenuBundle 1.2.0

So it seems after your explanation that current stable versions of phpcr-utils and MenuBundle are incompatible

**Comment by dbu** [comment]
oh, thats not good. i guess we should backport that fix and release 1.2.1 then.

**Comment by dbu** [comment]
actually the fix even was in the 1.2 branch, but there was no 1.2 release containing it. i created https://github.com/symfony-cmf/MenuBundle/releases/tag/1.2.1 - please update to that version. thanks for reporting the problem and helping to identify the problem, I appreciate it!

**Comment by nettiopsu** [comment]
Thanks for that fix!
